external/solr/src/main/java/org/apache/stormcrawler/solr/persistence/StatusUpdaterBolt.java

```diff
@@ -83,6 +126,23 @@ public void store(
             doc.setField(String.format(Locale.ROOT, "%s.%s", mdPrefix, key), values);
         }
 
+        String partitionKey = partitioner.getPartition(url, metadata);
+        if (partitionKey == null) {
+            partitionKey = "_DEFAULT_";
+        }
+
+        // store routing key in metadata?
+        if (StringUtils.isNotBlank(fieldNameForRoutingKey) && routingFieldNameInMetadata) {
```

Comment:
given that this will not change from one document to the other, why not determine the field name to use once and for all instead of doing it for every document?

Reply:
Yes, i will change this. I followed the logic from opensearch StatusUpdaterBolt that does this check inside `store()` but I guess this improvement can also be applied there.